clippy_lints/src/redundant_async_block.rs

```diff
@@ -42,12 +42,17 @@ impl EarlyLintPass for RedundantAsyncBlock {
         if expr.span.from_expansion() {
             return;
         }
-        if let ExprKind::Async(_, _, block) = &expr.kind && block.stmts.len() == 1 &&
+        if let ExprKind::Async(capture, _, block) = &expr.kind && block.stmts.len() == 1 &&
             let Some(Stmt { kind: StmtKind::Expr(last), .. }) = block.stmts.last() &&
             let ExprKind::Await(future) = &last.kind &&
             !future.span.from_expansion() &&
             !await_in_expr(future)
         {
+            if matches!(capture, CaptureBy::Value) && captured_by_ref(last) {
```

Comment:
This fix is rather incomplete because `CaptureBy::Ref` can also move a value into the closure:

```rust
use std::future::Future;

async fn work(_: &str) {}

fn spawn(_: impl Future<Output = ()> + 'static) {}

fn main() {
    let val = "Hello World".to_owned();
    spawn(async { work(&{ val }).await }); // false-positive
}
```

Reply:
Indeed, fixed.